src/utils/decoders.ts

```diff
@@ -140,9 +148,9 @@ export async function constructDataFromBytes(
 					data: {} // for consistency with legacy data format.
 				} as EthereumParsedData;
 				action =
-					firstByte === '00' || firstByte === '01'
+					firstByte === CRYPTO_ED25519 || firstByte === CRYPTO_SR25519
 						? 'signData'
-						: firstByte === '01'
+						: firstByte === CRYPTO_SR25519
```

Comment:
How did this condition ever get hit before? If `firstByte` is equal to `'01'`, then it would've gone through the first condition and `action` would be `'signedData'`.

Reply:
you're right, there's something fishy here. It's def. out of scope for this PR, but we should take a look and see if action being `signTransaction` is ever relevant. I guess it's not.

Reply:
Exactly; looks like nobody uses Ethereum support in Signer anymore. I think it should be removed completely and soon.